docs/cookbook/generating_synthetic_data.md

```diff
@@ -0,0 +1,240 @@
+    """
+    Generate {num_datapoints} random but realistic datapoints of items which could
+    be in a home appliances warehouse. Output the datapoints as a csv, and your
+    response should only be the CSV.
+
+    Format:
+    Name, Price, Inventory
+
+    Name - the name of the home appliance product
+    Price - the price of an individual product, in dollars (include cents)
+    Inventory - how many are left in stock
+    """
+)
+def generate_csv_data(num_datapoints: int): ...
+
+print(generate_csv_data(5))
+# > ```csv
+#   Name,Price,Inventory
+#   Stainless Steel Microwave,149.99,25
+#   Cordless Vacuum Cleaner,199.99,15
+#   4-Slice Toaster,39.99,50
+#   Energy Star Dishwasher,699.00,10
+#   Smart Air Purifier,249.50,8
+#   ```
+```
+
+Note that the prices and inventory of each item are somewhat realistic for their corresponding item, something which would be otherwise difficult to accomplish.
+
+## Generate Data with `response_model`
+
+Sometimes, it will be easier to integrate your datapoints into your code if they are defined as some schema, namely a Pydantic `BaseModel`. In this case, describe each column as the `description` of a `Field` in the `BaseModel` instead of the prompt, and set `response_model` to your defined schema:
+
+```python
+from pydantic import BaseModel, Field
+
+class HomeAppliance(BaseModel):
+    name: str = Field(description="The name of the home appliance product")
+    price: float = Field(
+        description="The price of an individual product, in dollars (include cents)"
+    )
+    inventory: int = Field(description="How many of the items are left in stock")
+
+@openai.call(model="gpt-4o-mini", response_model=list[HomeAppliance])
+@prompt_template(
+    """
+    Generate {num_datapoints} random but realistic datapoints of items which could
+    be in a home appliances warehouse. Output the datapoints as a list of instances
+    of HomeAppliance.
+    """
+)
+def generate_home_appliance_data(num_datapoints: int): ...
+
+print(generate_home_appliance_data(5))
+# > name='Refrigerator' price=799.99 inventory=15
+#   name='Microwave Oven' price=129.99 inventory=30
+#   name='Washing Machine' price=499.99 inventory=10
+#   name='Air Conditioner' price=349.99 inventory=5
+#   name='Dishwasher' price=599.99 inventory=8
+```
+
+## Generate Data into a pandas `DataFrame`
+
+Since pandas is a common library for working with data, it’s also worth knowing how to directly create and append to a dataframe with LLMs.
+
+### Create a New `DataFrame`
+
+To create a new `DataFrame`, we define a `BaseModel` schema with a simple function to generate  `DataFrame` via a list of list of data and the column names. One might be tempted to use a tool call, which at first might appear to be simpler, but would be bad practice since a tool should pass back a string so it may be passed back into the chat. Note that there are many ways to initialize `DataFrame`s and this is just one of the many ways it can be implemented:
```

Comment:
I'm not tempted to use a tool call here lol, I would just remove that whole section since tools seem obviously separate from data generation at least to me.